security/pkg/k8s/tokenreview/k8sauthn.go

```diff
@@ -170,3 +184,31 @@ func (authn *K8sSvcAcctAuthn) ValidateK8sJwt(targetToken string) ([]string, erro
 
 	return []string{namespace, saName}, nil
 }
+
+// isTrustworthyJwt checks if a jwt is a trustworthy jwt type.
+func isTrustworthyJwt(jwt string) (bool, error) {
```

Comment:
We need to also verify the audience of the JWT is Citadel. But that involves more changes including the Envoy container token volume mount. Let's defer that work.

Reply:
Sounds good. We can discuss this work offline.

Reply:
Yes, please add a TODO here. Thanks.